packages/arb-avm-cpp/avm/src/machinestate/machinestate.cpp

```diff
@@ -29,13 +30,24 @@ namespace {
 uint256_t max_arb_gas_remaining = std::numeric_limits<uint256_t>::max();
 }  // namespace
 
-AssertionContext::AssertionContext(
-    std::vector<Tuple> inbox_messages,
-    const nonstd::optional<uint256_t>& min_next_block_height,
-    uint256_t messages_to_skip)
-    : inbox_messages(std::move(inbox_messages)),
-      next_block_height(min_next_block_height),
-      inbox_messages_consumed(messages_to_skip) {}
+AssertionContext::AssertionContext(const MachineExecutionConfig& config)
```

Comment:
I think we'll want to take `MachineExecutionConfig` by value here so we can copy out the inbox messages unless there's a good reason to do otherwise